src/checksum.rs

```diff
@@ -0,0 +1,137 @@
+use modulo::Mod;
+
+// #[derive(Debug)]
+// pub struct Checksum {
+//     pub(crate) value: u8,
+// }
+
+pub type Checksum = u8;
+
+/// Calculate the checksum of some data
+pub fn calculate(data: &[u8]) -> Checksum {
+    let mut checksum = 0u8;
+
+    data.iter().for_each(|x| {
+        checksum = (checksum.overflowing_add(*x)).0.modulo(255);
```

Comment:
I could very well be mistaken, but: could this use the built in remainder `%` operator instead of the external modulo method? For an unsigned integer, [they should be equivalent I think](https://stackoverflow.com/a/13683709)

Reply:
the modulo crate is actually super straight forward... Sure we could replace it by our own implementation but that would be more of a copy/paste vs using a crate. There will not be much of a benefit IMO.
Once we have more tests and benchmarks though, those are some tests we can run to see if there is a benefit.

Reply:
Oh it's definitely not worth implementing the crate, I'm just wondering if

    use modulo::Mod;
    checksum = (checksum.overflowing_add(*x)).0.modulo(255);

could be replaced with [`std::op::Rem`](https://doc.rust-lang.org/std/ops/trait.Rem.html) in this case, e.g

    checksum = (checksum.overflowing_add(*x)).0 % 255;

Reply:
So I ran some tests and benchmarks with the code below.

<details>
<summary>tests & benchmarks</summary>
```

    #[test]
    fn test_checksum_crate_vs_alt() {
        let data = vec![0x0d, 0x0a, 0xf0, 0x0f, 0xff, 0x55];
        let checksum1 = calculate(&data);
        let checksum2 = calculate_alt(&data);
        assert_eq!(checksum1, checksum2);
    }

    #[bench]
    fn test_checksum_crate(b: &mut Bencher) {
        let data = vec![0x0d, 0x0a, 0xf0, 0x0f, 0xff, 0x55];
        let checksum = calculate(&data);
        assert_eq!(checksum, 150);
        b.iter(|| calculate(&data));
    }

    #[bench]
    fn test_checksum_no_crate(b: &mut Bencher) {
        let data = vec![0x0d, 0x0a, 0xf0, 0x0f, 0xff, 0x55];
        let checksum = calculate_alt(&data);
        assert_eq!(checksum, 150);
        b.iter(|| calculate(&data));
    }
```
</details>

The result of the benchmarks is:
```
test checksum::benchmarkss::test_checksum_crate    ... bench:           6 ns/iter (+/- 0)
test checksum::benchmarkss::test_checksum_no_crate ... bench:           6 ns/iter (+/- 0)
```

So in short:
- it works just as well without the extra crate
- the timing is the same in both cases

As conclusion, I don't think it makes sense using an extra crate when we don't need to. I will remove it.